packages/material-ui-lab/src/TreeItem/TreeItem.test.js

```diff
@@ -1055,6 +1069,18 @@ describe('<TreeItem />', () => {
           expect(getByTestId('one')).to.have.attribute('aria-selected', 'true');
         });
 
+        it('should deselect a node when clicking a selected node', () => {
```

Comment:
We have had a chat about this very problem with @dtassone on the Data Grid.

Do you have data that support this should be the default behavior?

Reply:
It seems that the current behavior is already correct. If people want this unselect behavior, isn't it a checkbox they are looking after instead?

This is the conclusion we had with the data grid.

Reply:
What about we only allow it when the ctrl/meta key is pressed? https://muix-storybook.netlify.app/?path=/docs/1-docs-4-selection--page

Reply:
VSCode's explorer (file system tree) keeps the clicked node selected. Furthermore it de-selects all other currently selected nodes (which were previously selected with CTRL-click). 

I tend to agree that de-selection does not make sense by default (but I'm only thinking about file trees right now). Usually it means that you open a file but in this case every click would keep the file open but every second time the node would not appear selected. That wouldn't be right.

Reply:
I think it depends on the context. If the tree is used on its own then de-selecting on click makes sense. But for the file explorer in an editor it makes sense since the click already has an action associated with it (open the file).

What about having it toggle the selection unless `.defaultPrevented`? Then an editor file explorer would call `event.preventDefault` in the click handler if it opens a file on click. Similar to how you would `event.preventDefault` to prevent navigation when you had an anchor (`<a />`) that was used as an action.

Reply:
We could also keep deselection as a controlled only feature and have a demo showing how to implement it. I have no preference.

Reply:
What about we study how are tree views used in the wild to make the call?

Reply:
I imagine that most tree views that allow deselection use checkboxes. I can't really think of a tree view that allows deselection when one node is selected.